File: app/src/main/java/com/arbaelbarca/listmovieskatalog/presenter/listmoviespresenter/ListMoviePresenter.java
package com.arbaelbarca.listmovieskatalog.presenter.listmoviespresenter;

import com.arbaelbarca.listmovieskatalog.model.movies.ResultsItem;
import com.arbaelbarca.listmovieskatalog.view.MainContract;

import java.util.ArrayList;

public class ListMoviePresenter implements MainContract.presenter, MainContract.GetNoticeIntractor.OnFinishedListener {
    private MainContract.MainView mainView;
    private MainContract.GetNoticeIntractor getNoticeIntractor;

    public ListMoviePresenter(MainContract.MainView mainView, MainContract.GetNoticeIntractor getNoticeIntractor) {
        this.mainView = mainView;
        this.getNoticeIntractor = getNoticeIntractor;
    }

    @Override
    public void onDestroy() {
        mainView = null;
    }


    @Override
    public void requestFromDataServer() {
        getNoticeIntractor.getNoticeArrayList(this, "popular");
    }

    @Override
    public void refreshData(String type) {
        getNoticeIntractor.getNoticeArrayList(this, type);

    }


    @Override
    public void onFinished(ArrayList<ResultsItem> resultsItemArrayList) {
        if (mainView != null) {
            mainView.setDataToRecyclerView(resultsItemArrayList);
            mainView.hideProgress();
        }
    }

    @Override
    public void onFailure(Throwable t) {
        if (mainView != null)
            mainView.onResponseFailure(t);
    }
}

File: app/src/main/java/com/robytakwa/listpicturekatalog/presenter/listmoviespresenter/ListMoviePresenter.kt
package com.robytakwa.listpicturekatalog.presenter.listmoviespresenter

import com.robytakwa.listpicturekatalog.model.movies.ResultsItem
import com.robytakwa.listpicturekatalog.view.MainContract
import java.util.*

class ListMoviePresenter(private var mainView: MainContract.MainView?, private val getNoticeIntractor: MainContract.GetNoticeIntractor) : MainContract.presenter, MainContract.GetNoticeIntractor.OnFinishedListener {

    override fun onDestroy() {
        mainView = null
    }


    override fun requestFromDataServer() {
        getNoticeIntractor.getNoticeArrayList(this, "popular")
    }

    override fun refreshData(type: String) {
        getNoticeIntractor.getNoticeArrayList(this, type)

    }


    override fun onFinished(resultsItemArrayList: ArrayList<ResultsItem>) {
        if (mainView != null) {
            mainView!!.setDataToRecyclerView(resultsItemArrayList)
            mainView!!.hideProgress()
        }
    }

    override fun onFailure(t: Throwable) {
        if (mainView != null)
            mainView!!.onResponseFailure(t)
    }
}
